app/utils/metadata.py

```diff
@@ -568,506 +568,280 @@ def __update_from_settings(self) -> None:
-                                dependencies_by_ver["li"],
-                                self.internal_local_metadata,
-                            )
-                        else:
-                            logger.warning(
-                                f"About.xml syntax error. Unable to read <moddependenciesbyversion> tag from XML for version [{version}]: {self.internal_local_metadata[uuid]['metadata_file_path']}"
-                            )
-                            logger.debug(dependencies_by_ver)
-            if self.internal_local_metadata[uuid].get("incompatiblewith"):
-                incompatibilities = self.internal_local_metadata[uuid][
-                    "incompatiblewith"
-                ].get("li")
-                if incompatibilities:
-                    logger.debug(
-                        f"Current mod is incompatible with these mods: {incompatibilities}"
-                    )
-                    add_incompatibility_to_mod(
-                        self.internal_local_metadata[uuid],
-                        incompatibilities,
-                        self.internal_local_metadata,
-                    )
+        def process_incompatibilities(uuid, incompatibilities):
+            if incompatibilities:
+                logger.debug(
+                    f"Current mod is incompatible with these mods: {incompatibilities}"
+                )
+                add_incompatibility_to_mod(
+                    self.internal_local_metadata[uuid],
+                    incompatibilities,
+                    self.internal_local_metadata,
+                )
 
-            if self.internal_local_metadata[uuid].get("incompatiblewithbyversion"):
-                major, minor = self.game_version.split(".")[
-                    :2
-                ]  # Split the version and take the first two parts
-                version_regex = rf"v{major}\.{minor}"  # Construct the regex to match both major and minor versions
-                for version, incompatibilities_by_ver in self.internal_local_metadata[
-                    uuid
-                ]["incompatiblewithbyversion"].items():
+        def process_load_order(uuid, load_these, rule_type, opposite_rule_type):
+            if load_these:
+                logger.debug(f"Current mod should {rule_type} these mods: {load_these}")
+                add_load_rule_to_mod(
+                    self.internal_local_metadata[uuid],
+                    load_these,
+                    rule_type,
+                    opposite_rule_type,
+                    self.internal_local_metadata,
+                    self.packageid_to_uuids,
+                )
+
+        for uuid in uuids:
+            mod_metadata = self.internal_local_metadata[uuid]
+            logger.debug(f"UUID: {uuid} packageid: {mod_metadata.get('packageid')}")
+
+            # Process moddependencies
+            dependencies = mod_metadata.get("moddependencies")
+            if dependencies:
```

Comment:
Minor thing, and I know we didn't have this before, but as an overall improvement to RimSort we should start being explicit with conditionals involving `None`s. Since `dependencies` could be anything here, including an empty string, empty list, etc, `if dependencies...` could resolve to `True` even though we got something erroneous back.

This is common across RimSort but we should try to change it

Reply:
i think i fixed it now with rewrite